### ISSUE TITLE
Implement Attach Files to discussion form

### ISSUE THREAD

**Issue by blesildaramirez** [pkp/pkp-lib#12117]
## Description:
Implement the Attach Files feature in the discussion form.

## Requirements:
- [x] When creating a new task/discussion or adding a new message, the user should be able to attach files.
- [x] The following options should be available when attaching files:
  - [x] Upload file from computer
  - [x] Workflow Files
  - [ ] ~~Library Files~~ (for consideration)
- [x] Files selected from the modal should appear in the **Attached Files** list below the discussion text area.
- [x] When reopening the **Attach Files** modal, all previously selected files must have their checkboxes pre-selected (ticked).
- [x] Validations:
  - [x] Attached files should be saved against the message/headnote. When editing the task/discussion, the attached files could also be updated. 

## Access Rules: 🚀 
- [x] To retain our validations with the current implementation, we will not allow authors and reviewers to attach from workflow files.
- [x] For other editorial roles, only the accessible files for the selected stage are available.

**What application are you using?**
  OJS, OMP or OPS main branch

**Additional information**
- Mockups can be found in this link: https://miro.com/app/board/uXjVLjmM3rU=/?moveToWidget=3458764622463205023&cot=14
- An existing submission file in the workflow stage will be re-uploaded when attached in the headnote/notes

---
## Development Updates
✅ = merged
⌛ = work in progress
📌 = ready for code review

**Pull requests**
ui-library:
✅ https://github.com/pkp/ui-library/pull/749 Implement Attach Files (Workflow Files)
📌 https://github.com/pkp/ui-library/pull/765 Implement searching of suggested templates and update access rules for edit and delete, integrate with attach files api

pkp-lib:
✅ https://github.com/pkp/pkp-lib/pull/12103 Add locale keys for attach files component

**Comment by Tribunal33** [comment]
This is looking good @blesildaramirez.  Let me know when it is complete and I can do some more testing around those areas that are still WIP.  

Here are the test scenarios I have so far:
1. Test Task and Discussions Attach Files is in the comments field
2. Test Upload files button
3. Test Upload files option drag and drop onto area
4. Test Upload files option click and upload file
5. Test Upload files option Add files button
6. Test that the users who participate in the task can see the attached files (ignored)
7. Test Workflow files button
8. Test Workflow files stages display the correct files from each stage
9. Test workflow files add files from multiple stages
10. Test Adding Workflow files and Upload files from computer to same task/discussion
11. Test Workflow files add files 3 dots More information
12. Test Workflow files add files 3 dots More information History
13. Test workflow files add files 3 dots More information Notes
14. Test Library Files (ignored)
15. Test Validation Attached files should be saved against the message/headnote (ignored)

**Comment by Tribunal33** [comment]
@blesildaramirez Found one issue when a section editor, author or reviewer tries to respond to a discussion with files attached from JM.  Let me know if you want me to write up a bug for it.  Not sure if this was ready for this type of test yet.

**Update:** I might be wrong. looks like it is any reply to a discussion that has a file when there is no other file attached.  So replying without attaching a file is throwing the error.   I can reply to a discussion with a file attached without error.

Steps to reproduce are:

1. Login with Dbarnes (Any user)
2. Find submission in any stage
3. Add discussion with attached files
4. Login with dbuskins (section editor) or author or reviewer -- or any user
5. Reply to the discussion but do not attach a new file -> click save
6. Error seen

<img width="1202" height="561" alt="Image" src="https://github.com/user-attachments/assets/10e8834b-93a6-40e1-8b61-41ae12edec9e" />

Another related issue (I think).  A different error came up when trying to add to a discussion with an attached workflow file.  No logs for this one as it wasn't a 500 error. 

Steps to reproduce are: (almost the same)
1.  Loging with an editor
2. Find a submission in review stage with revisions (can repro without revisions too)
3. Add a discussion with an attached file from workflow.  > Save should be successful.
4. Add a new response to the discussion with another attached file from workflow
5. Try to save the response

<img width="2200" height="519" alt="Image" src="https://github.com/user-attachments/assets/f07c82e4-2ded-45fe-8c77-a75ad2a64a62" />

**Comment by Tribunal33** [comment]
Not sure about this requirement but should it be duplicating the files when referencing the workflow files.  Additionally, when the comment is opened/edited the ID is missing from the attached files and makes it a bit confusing as to what file is what.  

Anytime another user creates a comment and references an workflow file it will then show this duplicated file in the workflow list with a new ID.  But there is only one original or revised file in the submission table.
<img width="1975" height="892" alt="Image" src="https://github.com/user-attachments/assets/7e4477c8-77c3-4210-8f4e-bf5998688541" />

Here is the missing ID reference portion.

<img width="1552" height="232" alt="Image" src="https://github.com/user-attachments/assets/347a179e-1ce6-4e54-86d4-d6a5ee25de8a" />

**Comment by blesildaramirez** [comment]
Hi @Tribunal33 ,
For the first issue, this happened because the `temporaryFileIds` is set to empty array `[]`, which should be `undefined` so the API will not look for any IDs in the temporary files table. It should now be fixed when you test again using the PR-5222 VM.

For the 500 error, can you recheck this? I can't seem to replicate it, it might've been included in the fix above.

Regarding duplicated files - this is expected as new notes create new copies of the attached files in the workflow files. We can raise this one up to @Vitaliy-1  or @Devika008 to get their thoughts on the duplication.

For the missing file id in the attached file when in message list, this is now fixed. I added the file ID to be displayed with the attached files. Although this wasn't included in the mockup, for now let's keep the ID visible so you can verify it during your testing.

**Comment by Tribunal33** [comment]
All above issues have been resolved.  This passes QA.  

Here are the testing scenarios
1. Test Task and Discussions Attach Files is in the comments field
2. Test Upload files button
3. Test Upload files option drag and drop onto area
4. Test Upload files option click and upload file
5. Test Upload files option Add files button
6. Test that the users who participate in the task can see the attached files
7. Test Workflow files button
8. Test Workflow files stages display the correct files from each stage
9. Test workflow files add files from multiple stages
10. Test Adding Workflow files and Upload files from computer to same task/discussion
11. Test Workflow files add files 3 dots More information
12. Test Workflow files add files 3 dots More information History
13. Test workflow files add files 3 dots More information Notes
14. ~~Test Library Files~~ (out of scope)
15. Test Validation Attached files should be saved against the message/headnote
16. Test that when an Author tries to attach files that workflow files are not an option
17. Test that when a Reviewer tries to attach files that workflow files are not an option
18. Test that Author, Reviewer, Section editor can respond when there are files attached
19. Test other editorial roles can only access workflow files from specific stages that they have access too
20. Test that adding a revised file does not duplicate the file references
